src/compiler/types.ts

```diff
@@ -1839,6 +1841,11 @@ namespace ts {
         numberIndexType?: Type;            // Numeric index type
     }
 
+    /* @internal */
+    export interface FreshObjectLiteralType extends ResolvedType {
+        regularType: ResolvedType;  // Regular version of fresh type

```

Comment:
They are exactly the same, except the TypeFlags.FreshObjectLiteral flag is set in the fresh version.